src/main/java/nus/climods/logic/commands/AddCommand.java

```diff
@@ -22,42 +25,57 @@ public class AddCommand extends Command {
     public static final String MESSAGE_MODULE_NOT_FOUND = "Module not in current NUS curriculum";
     public static final String MESSAGE_MODULE_NOT_OFFERED_IN_SEMESTER = "Module not offered in chosen semester";
 
-    private final String moduleCode;
+    private final String toAdd;
     private final SemestersEnum semester;
 
     /**
      * Creates an AddCommand to add the specified {@code Module}
      */
-    public AddCommand(String moduleCode, SemestersEnum semester) {
-        requireNonNull(moduleCode);
-        this.moduleCode = moduleCode.toUpperCase();
+    public AddCommand(String toAdd, SemestersEnum semester) {
+        requireNonNull(toAdd);
+        this.toAdd = toAdd.toUpperCase();
         this.semester = semester;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        if (!model.isModuleOffered(moduleCode)) {
+        if (!model.isModuleOffered(toAdd)) {
             throw new CommandException(MESSAGE_MODULE_NOT_FOUND);
         }
-        if (!model.isModuleOfferedInSemester(moduleCode, semester)) {
+        if (!model.isModuleOfferedInSemester(toAdd, semester)) {
             throw new CommandException(MESSAGE_MODULE_NOT_OFFERED_IN_SEMESTER);
         }
 
-        UserModule moduleToAdd = new UserModule(moduleCode, semester);
+        UserModule moduleToAdd = new UserModule(toAdd, semester);
         if (model.hasUserModule(moduleToAdd)) {
             throw new CommandException(MESSAGE_DUPLICATE_MODULE);
         }
 
+        Module module = model.getListModule(toAdd).get();
+
+        try {
+            module.loadMoreData();
+        } catch (ApiException e) {
+            throw new CommandException("Error 404 something went wrong");
```

Comment:
There's a `getCode()` in ApiException, can use it to find out the actual error code

Reply:
Also, I found a bug where this command works

![image](https://user-images.githubusercontent.com/53803800/198358576-c3139e6c-71f8-4058-99a2-007cbbfacda5.png)

I will open an issue and this can be resolve in v1.4